src/diag/diag_process.cpp

```diff
@@ -309,6 +310,47 @@ void Diag::ProcessStats(int aArgCount, char *aArgVector[], char *aOutput, size_t
     AppendErrorResult(error, aOutput, aOutputMaxLen);
 }
 
+void Diag::ProcessRadio(int aArgCount, char *aArgVector[], char *aOutput, size_t aOutputMaxLen)
+{
+    otError error = OT_ERROR_INVALID_ARGS;
+
+    VerifyOrExit(otPlatDiagModeGet(), error = OT_ERROR_INVALID_STATE);
+    VerifyOrExit(aArgCount > 0, error = OT_ERROR_INVALID_ARGS);
+
+    if (strcmp(aArgVector[0], "sleep") == 0)
+    {
+        if (otPlatRadioIsEnabled(sInstance))
+        {
+            snprintf(aOutput, aOutputMaxLen, "radio is not enable now\r\n");
+        }
+        else
+        {
+            error = otPlatRadioSleep(sInstance);
+            snprintf(aOutput, aOutputMaxLen, "set radio from receive to sleep \r\nstatus 0x%02x\r\n", error);
+        }
+    }
+    else if (strcmp(aArgVector[0], "receive") == 0)
+    {
+        if (aArgCount > 1)
+        {
+            long value;
+
+            SuccessOrExit(error = ParseLong(aArgVector[1], value));
```

Comment:
We already have `diag channel`, I suggest keep using that. Would you like submit another PR to add the missing documents for `diag channel`!

Reply:
It could be.. But "diag channel" return void; It may need to change diag channel function. Is that OK?